features/storage/blockdevice/ChainingBlockDevice.cpp

```diff
@@ -283,4 +283,9 @@ bd_size_t ChainingBlockDevice::size() const
     return _size;
 }
 
+const char * ChainingBlockDevice::get_type()
+{
+    return "CHAINING";
```

Comment:
Chaining is a special case and is not covered in design doc,

Question - Do we allow different type of `physical` block devices to be chained together?

Reply:
Chaining different block devices is currently allowed (there's no way to check such a thing up to this PR). However, the more interesting question is whether it *should* be allowed. I don't have a definite answer for it. However, if we wish to forbid it, we can change the type in this PR to the underlying one's.
@geky What do you say?

Reply:
> Chaining different physical block devices is currently allowed

Thanks answers my query

> However, the more interesting question is whether it should be allowed.

Yes, but can keep it outside the scope of this PR.

For this PR, we should document the special case of chaining as note in API. Clearly stating, that instead of physical block device, response return value is "chaining" string

Reply:
You are right about the documentation. I will do it tomorrow morning.

Reply:
Documentation for ChainingBlockDevice added